skimage/filters/rank/core_cy.pyx

```diff
@@ -126,115 +126,116 @@ cdef void _core(void kernel(dtype_t_out*, Py_ssize_t, Py_ssize_t[::1], double,
 
     num_se_n = num_se_s = num_se_e = num_se_w = 0
 
-    for i in range(n_bins):
-        histo[i] = 0
-
-    for r in range(srows):
-        for c in range(scols):
-            if t_e[r, c]:
-                se_e_r[num_se_e] = r - centre_r
-                se_e_c[num_se_e] = c - centre_c
-                num_se_e += 1
-            if t_w[r, c]:
-                se_w_r[num_se_w] = r - centre_r
-                se_w_c[num_se_w] = c - centre_c
-                num_se_w += 1
-            if t_n[r, c]:
-                se_n_r[num_se_n] = r - centre_r
-                se_n_c[num_se_n] = c - centre_c
-                num_se_n += 1
-            if t_s[r, c]:
-                se_s_r[num_se_s] = r - centre_r
-                se_s_c[num_se_s] = c - centre_c
-                num_se_s += 1
-
-    for r in range(srows):
-        for c in range(scols):
-            rr = r - centre_r
-            cc = c - centre_c
-            if selem[r, c]:
+    with nogil:
```

Comment:
I don't think this should be necessary since the function itself is declared nogil above:
https://github.com/scikit-image/scikit-image/blob/54e424a7806f2545665f4f1fbd154133504f0370/skimage/filters/rank/core_cy.pyx#L50

Reply:
nevermind, that `nogil` I cited was on the nested kernel function, not the outer _core. sorry for the mixup!